src/mixins/dpValidateMixin.js

```diff
@@ -29,20 +29,30 @@ export default {
       const isForm = this.$el.hasAttribute('data-dp-validate') && this.$el.getAttribute('data-dp-validate') === formId
       const form = isForm ? this.$el : this.$el.querySelector(`[data-dp-validate=${formId}]`)
       const formValidation = validateForm(form)
-      const newValidateObject = { ...this.dpValidate, [formId]: formValidation.valid }
-      this.dpValidate = newValidateObject
+      this.dpValidate = { ...this.dpValidate, [formId]: formValidation.valid }
 
       if (hasOwnProp(this.dpValidate, 'invalidFields') === false) {
         this.dpValidate.invalidFields = {}
       }
       this.dpValidate.invalidFields[formId] = formValidation.invalidFields
+
       if (this.dpValidate[formId] === false) {
         const customErrors = this.dpValidate.invalidFields[formId]
           .filter(element => element.hasAttribute('data-dp-validate-error'))
           .map(element => element.dataset.dpValidateError)
         customErrors.forEach(error => dplan.notify.notify('error', Translator.trans(error)))
+
         if (customErrors.length === 0) {
-          dplan.notify.notify('error', Translator.trans('error.mandatoryfields.no_asterisk'))
+          const invalidFields = this.dpValidate.invalidFields[formId]
+          const missingFieldNames = invalidFields.map(field => field.getAttribute('data-dp-validate-error-fieldname') || '')
+          const nonEmptyFieldNames = missingFieldNames.filter(el => el !== '')
```

Comment:
As [getAttribute defaults to null or empty string](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute), we could simplify the code like so:

```suggestion
          const nonEmptyFieldNames = invalidFields
            .map(field => field.getAttribute('data-dp-validate-error-fieldname'))
            .filter(Boolean)
```

Maybe using the [Boolean Constructor as a callback](https://mikebifulco.com/posts/javascript-filter-boolean) is too much magic but i guess it works.

Reply:
thx, done here https://github.com/demos-europe/demosplan-ui/pull/359/commits/aa52da6e7983653787044b4ebc881e065b739532